src/test/java/com/kodilla/ecommercee/domain/CartEntityTest.java

```diff
@@ -0,0 +1,96 @@
+package com.kodilla.ecommercee.domain;
+
+import com.kodilla.ecommercee.repository.CartRepository;
+import com.kodilla.ecommercee.repository.ProductsInCartRepository;
+import com.kodilla.ecommercee.repository.UserRepository;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+@SpringBootTest
+public class CartEntityTest {
+
+    @Autowired
+    CartRepository cartRepository;
+    @Autowired
+    UserRepository userRepository;
+    @Autowired
+    ProductsInCartRepository productsInCartRepository;
+
+    private Cart createCart(){
+        Cart cart = new Cart();
+        User user = new User();
+        List<ProductsInCart> productsInCartList = new ArrayList<>();
+
+        ProductsInCart productsInCart1 = new ProductsInCart();
+        ProductsInCart productsInCart2 = new ProductsInCart();
+        productsInCartList.add(productsInCart1);
+        productsInCartList.add(productsInCart2);
+
+        productsInCartRepository.save(productsInCart1);
+        productsInCartRepository.save(productsInCart2);
+
+        userRepository.save(user);
+
+        cart.setUser(user);
+        cart.setProductsInCart(productsInCartList);
+
+        return cart;
+    }
+
+    @Test
+    void testSaveCart(){
+        //Given
+        Cart cart = createCart();
+        //When
+        cartRepository.save(cart);
+        Long cartId = cart.getCartId();
+        //Then
+        assertNotNull(cartId);
```

Comment:
Don't see assertion for produtsInCart, it's your related entity so you should check is it exists.